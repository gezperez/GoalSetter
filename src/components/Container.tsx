import React, { FC, ReactNode } from 'react'
import { StatusBar, StyleProp, StyleSheet, View, ViewStyle } from 'react-native'
import { ScrollView } from 'react-native-gesture-handler'
import { useSafeAreaInsets } from 'react-native-safe-area-context'

type ContainerProps = {
  children?: ReactNode
  contentContainerStyle?: StyleProp<ViewStyle>
  accessibilityLabel?: string
  testID?: string
}

const Container: FC<ContainerProps> = ({ children, contentContainerStyle, testID, accessibilityLabel }) => {
  const insets = useSafeAreaInsets()

  return (
    <View
      style={[styles.container, { paddingBottom: insets.bottom }]}
      testID={testID}
      accessibilityLabel={accessibilityLabel}
    >
      <StatusBar barStyle={'light-content'} translucent backgroundColor={'transparent'} />
      <ScrollView contentContainerStyle={[contentContainerStyle, styles.scrollView]}>{children}</ScrollView>
    </View>
  )
}

type Styles = {
  container: ViewStyle
  scrollView: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  container: {
    height: '100%',
  },
  scrollView: {
    flexGrow: 1,
  },
})

export default Container
