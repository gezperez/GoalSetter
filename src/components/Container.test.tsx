import React from 'react'
import { View } from 'react-native'
import { SafeAreaProvider } from 'react-native-safe-area-context'
import { render } from '@testing-library/react-native'

import Container from './Container'

// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
// @ts-ignore
global.React = React

describe('Container', () => {
  afterEach(jest.resetAllMocks)

  it('should render with required props', () => {
    const { toJSON, getByTestId } = render(
      <SafeAreaProvider>
        <Container testID="container" />
      </SafeAreaProvider>
    )

    const container = getByTestId('container')

    expect(container).toBeDefined()
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render with children prop', () => {
    const { toJSON, getByTestId } = render(
      <SafeAreaProvider>
        <Container testID="container">
          <View />
        </Container>
      </SafeAreaProvider>
    )

    const container = getByTestId('container')

    expect(container).toBeDefined()
    expect(toJSON()).toMatchSnapshot()
  })
})
