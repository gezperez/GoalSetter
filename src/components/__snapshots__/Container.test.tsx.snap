// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render with children prop 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="container"
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Array [
        undefined,
        Object {
          "flexGrow": 1,
        },
      ]
    }
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View />
    </View>
  </RCTScrollView>
</View>
`;

exports[`Container should render with required props 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="container"
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Array [
        undefined,
        Object {
          "flexGrow": 1,
        },
      ]
    }
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View />
  </RCTScrollView>
</View>
`;
